IBF-Typhoon-model/lib/pipeline.py

```diff
@@ -0,0 +1,314 @@
+            centroid_id = np.tile(centroid_idx, nsteps)
+            intensity_3d = windfield[0].toarray().reshape(nsteps, ncents, 2)
+            intensity = np.linalg.norm(intensity_3d, axis=-1).ravel()
+            timesteps = np.repeat(track.data[0].time.values, ncents)
+            #timesteps = np.repeat(tr.time.values, ncents)
+            timesteps = timesteps.reshape((nsteps, ncents)).ravel()
+            inten_tr = pd.DataFrame({
+                    'centroid_id': centroid_id,
+                    'value': intensity,
+                    'timestamp': timesteps,})
+            inten_tr = inten_tr[inten_tr.value > threshold]
+            inten_tr['storm_id'] = tr.sid
+            inten_tr['ens_id'] =tr.sid+'_'+str(tr.ensemble_number)
+            inten_tr['name'] = tr.name
+            list_intensity.append(inten_tr)
+            distan_track1=[]
+            for index, row in df.iterrows():
+                dist=np.min(np.sqrt(np.square(tr.lat.values-row['lat'])+np.square(tr.lon.values-row['lon'])))
+                distan_track1.append(dist*111)
+            dist_tr = pd.DataFrame({'centroid_id': centroid_idx,'value': distan_track1})
+            dist_tr['storm_id'] = tr.sid
+            dist_tr['name'] = tr.name
+            dist_tr['ens_id'] =tr.sid+'_'+str(tr.ensemble_number)
+            distan_track.append(dist_tr)                
+        df_intensity = pd.concat(list_intensity)
+        df_intensity=pd.merge(df_intensity, df_admin, how='outer', on='centroid_id')
+        df_intensity=df_intensity.dropna()
+        
+        df_intensity_=df_intensity.groupby(['adm3_pcode','ens_id'],as_index=False).agg({"value":['count', 'max']}) 
+        # rename columns
+        df_intensity_.columns = [x for x in ['adm3_pcode','storm_id','value_count','v_max']] 
+        distan_track1= pd.concat(distan_track)
+        distan_track1=pd.merge(distan_track1, df_admin, how='outer', on='centroid_id')
+        distan_track1=distan_track1.dropna()
+        
+        distan_track1=distan_track1.groupby(['adm3_pcode','name','ens_id'],as_index=False).agg({'value':'min'}) 
+        distan_track1.columns = [x for x in ['adm3_pcode','name','storm_id','dis_track_min']]#join_left_df_.columns.ravel()] 
+        typhhon_df = pd.merge(df_intensity_, distan_track1,  how='left', on=['adm3_pcode','storm_id']) 
+    
+        typhhon_df.to_csv(os.path.join(Input_folder,'windfield.csv'), index=False)
+    
+        line_='windfield,'+'%swindfield.csv' % Input_folder+ ',' +typhoons+','+ datetime.now().strftime("%Y%m%d%H")   #StormName #
+        #line_='Rainfall,'+'%sRainfall/' % Input_folder +','+ typhoons + ',' + datetime.now().strftime("%Y%m%d%H") #StormName #
+        fname.write(line_+'\n')
+        fname.close()
+        
+        
+        #############################################################
+        #### Run IBF model 
+        #############################################################
+        os.chdir(path)
```

Comment:
Better remove this line if you are including `path` as a command-line arg (which I assume is for the purpose of changing the location of the input and output files)